### PR TITLE
feat: caip-25 db migrations

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

**Review by Elyniss — APPROVED** [review]

File: sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt [hunk 1]
```diff
@@ -73,7 +73,37 @@ class ConnectViewModel : ViewModel() {
...
+                }.toMap()
+
+        val properties: Map<String, String> = mapOf("sessionExpiry" to "123456789")
+
+        val connectParams =
+            Sign.Params.Connect(namespaces = namespaces.toMutableMap().plus(tmp), optionalNamespaces = optionalNamespaces, properties = properties, pairing = pairing)
+
+        println("Kobe; Dapp: SessionNamespaces: ${connectParams.namespaces}")
+        println("Kobe; Dapp: OptionalNamespaces: ${connectParams.optionalNamespaces}")
+        println("Kobe; Dapp: Properties: ${connectParams.properties}")
```
Comment (Elyniss) [inline sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt:105]: Better to delete them

Comment (jakubuid) [reply sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt:105]: Yes yes I'm gonna remove them after cross-platform testing session

File: samples/web3wallet/src/main/kotlin/com/walletconnect/web3/wallet/ui/routes/dialog_routes/session_proposal/SessionProposalViewModel.kt [hunk 2]
```diff
@@ -16,46 +16,78 @@ class SessionProposalViewModel : ViewModel() {
...
+
+            val sessionNamespacesIndexedByNamespace: Map<String, Wallet.Model.Namespace.Session> =
+                selectedAccounts.filter { (chain: Chains, _) ->
+                    sessionProposal.requiredNamespaces
+                        .filter { (_, namespace) -> namespace.chains != null }
+                        .flatMap { (_, namespace) -> namespace.chains!! }
+                        .contains(chain.chainId)
+                }.toList()
+                    .groupBy { (chain: Chains, _: String) -> chain.chainNamespace }
+                    .map { (key: String, chainData: List<Pair<Chains, String>>) ->
```
Comment (TalhaAli00) [inline samples/web3wallet/src/main/kotlin/com/walletconnect/web3/wallet/ui/routes/dialog_routes/session_proposal/SessionProposalViewModel.kt:32]: use `associate`

File: sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt [hunk 3]
```diff
@@ -63,8 +63,8 @@ class ConnectViewModel : ViewModel() {
         }
         val namespaces: Map<String, Sign.Model.Namespace.Proposal> =
             listOfChainUI
-                .filter { it.isSelected }
-                .groupBy { it.chainNamespace }// OR chaiId
+                .filter { it.isSelected && it.chainId != "eip155:137" && it.chainId != "eip155:42" }
```
Comment (TalhaAli00) [inline sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt:66]: replace hardcoded chain id with `Chains` enum in samples common

File: sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt [hunk 4]
```diff
@@ -73,7 +73,38 @@ class ConnectViewModel : ViewModel() {
...
+            .map { (key: String, selectedChains: List<ChainSelectionUI>) ->
+                key to Sign.Model.Namespace.Proposal(
+                    methods = selectedChains.flatMap { it.methods }.distinct(),
+                    events = selectedChains.flatMap { it.events }.distinct()
+                )
+            }.toMap()
+
+        val optionalNamespaces: Map<String, Sign.Model.Namespace.Proposal> =
+            listOfChainUI
+                .filter { it.isSelected && it.chainId == "eip155:137" }
```
Comment (TalhaAli00) [inline sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt:89]: same

File: sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt [hunk 5]
```diff
@@ -73,7 +73,38 @@ class ConnectViewModel : ViewModel() {
                     )
                 }.toMap()
 
-        val connectParams = Sign.Params.Connect(namespaces = namespaces, pairing = pairing)
+
+        val tmp = listOfChainUI
+            .filter { it.isSelected && it.chainId == "eip155:42" }
```
Comment (TalhaAli00) [inline sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt:78]: same

File: sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt [hunk 6]
```diff
@@ -73,7 +73,38 @@ class ConnectViewModel : ViewModel() {
...
+                .filter { it.isSelected && it.chainId == "eip155:137" }
+                .groupBy { it.chainId }
+                .map { (key: String, selectedChains: List<ChainSelectionUI>) ->
+                    key to Sign.Model.Namespace.Proposal(
+                        methods = selectedChains.flatMap { it.methods }.distinct(),
+                        events = selectedChains.flatMap { it.events }.distinct()
+                    )
+                }.toMap()
+
+        val properties: Map<String, String> = mapOf("sessionExpiry" to "123456789")
```
Comment (TalhaAli00) [inline sign/dapp/src/main/kotlin/com/walletconnect/dapp/ui/connect/ConnectViewModel.kt:98]: use valid expiry

File: sign/wallet/src/main/kotlin/com/walletconnect/wallet/ui/host/proposal/SessionProposalViewModel.kt [hunk 7]
```diff
@@ -26,41 +26,68 @@ class SessionProposalViewModel : ViewModel() {
...
+                    }.toMap()
+
+            val sessionNamespacesIndexedByChain: Map<String, Sign.Model.Namespace.Session> =
+                selectedAccounts.filter { (chain: Chains, _) ->
+                    sessionProposal.requiredNamespaces
+                        .filter { (namespaceKey, namespace) -> namespace.chains == null && namespaceKey == chain.chainId }
+                        .isNotEmpty()
+                }.toList()
+                    .groupBy { (chain: Chains, _: String) -> chain.chainId }
+                    .map { (key: String, chainData: List<Pair<Chains, String>>) ->
```
Comment (TalhaAli00) [inline sign/wallet/src/main/kotlin/com/walletconnect/wallet/ui/host/proposal/SessionProposalViewModel.kt:70]: use associate

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**